mm2src/coins/utxo.rs

```diff
@@ -2649,3 +2649,69 @@ fn kmd_interest(height: Option<u64>, value: u64, lock_time: u64, current_time: u
     minutes -= 59;
     (value / 10_512_000) * minutes
 }
+
+fn kmd_interest_accrue_stop_at(height: u64, lock_time: u64) -> u64 {
+    let minutes = if height < 1_000_000 {
+        // interest stop accruing after 1 year before block 1000000
+        365 * 24 * 60
+    } else {
+        // interest stop accruing after 1 month past 1000000 block
+        31 * 24 * 60
+    };
+
+    lock_time + minutes
+}
+
+#[derive(Deserialize, Eq, PartialEq, Serialize)]
+pub struct KmdRewardInfoElement {
```

Comment:
Could you please also add the explanation of why rewards are not accruing (amount ==0)? There're just few possible reasons so we can use a enum for this, e.g.:
```
enum RewardsNotAccrueReason {
  LocktimeNotSet,
  UtxoAmountLessThanTen,
  ...
}
```

Reply:
I guess the **reason** field should be optional?

Reply:
Yes, it makes sense only when `KmdRewardInfoElement::amount == 0`.

Reply:
I also offer to separate the output structure to the enum:

```
#[serde(tag = "rewards_type")]
enum KmdOutputRewardsInfo {
    Accrue {
        tx_hash: H256Json,
        height: u64,
        amount: u64,
        locktime: u64,
        /// Amount of accrued rewards.
        accrued_rewards: u64,
        /// Rewards accruing stop at this time for the given transaction.
        accrue_stop_at: u64,
    },
    NotAccrue {
        tx_hash: H256Json,
        height: Option<u64>,
        amount: u64,
        locktime: Option<u64>,
        reason: KmdRewardsNotAccrueReason,
    },
}
```

Reply:
tx_hash, height, amount and locktime are independent from whether rewards are available or not. I guess it will be sufficient to have `enum RewardsAccrueInfo {
    Accrued(BigDecimal),
    LocktimeNotSet,
    etc.
}` for `accrued_rewards` field and keep the remaining structure of `KmdOutputRewardsInfo` same.

Reply:
Please also add `input_index` field because address can have multiple utxos in same tx.

Reply:
I considered to use another enum for the NotAccturedReason, because it avoids changing the kmd_interest() return type from u64 to BigDecimal.